client/state/billing-transactions/actions.ts

```diff
@@ -13,39 +13,59 @@ import type { CalypsoDispatch } from '../types';
 import 'calypso/state/billing-transactions/init';
 
 export const requestBillingTransactions = ( transactionType?: BillingTransactionsType ) => {
-	return ( dispatch: CalypsoDispatch ) => {
+	return async ( dispatch: CalypsoDispatch ) => {
 		dispatch( {
 			type: BILLING_TRANSACTIONS_REQUEST,
 		} );
 
-		return wp.req
-			.get( '/me/billing-history' + ( transactionType ? `/${ transactionType }` : '' ), {
+		const limit = 200;
+		let url = '/me/billing-history' + ( transactionType ? `/${ transactionType }` : '' );
+		if ( transactionType !== 'upcoming' ) {
+			url = url + '?limit=' + limit;
+		}
+
+		try {
+			const {
+				billing_history,
+				upcoming_charges,
+			}: {
+				billing_history: BillingTransaction[];
+				upcoming_charges: UpcomingCharge[];
+			} = await wp.req.get( url, {
 				apiVersion: '1.3',
-			} )
-			.then(
-				( {
-					billing_history,
-					upcoming_charges,
-				}: {
-					billing_history: BillingTransaction[];
-					upcoming_charges: UpcomingCharge[];
-				} ) => {
-					dispatch( {
-						type: BILLING_TRANSACTIONS_RECEIVE,
-						past: billing_history,
-						upcoming: upcoming_charges,
-					} );
-					dispatch( {
-						type: BILLING_TRANSACTIONS_REQUEST_SUCCESS,
+			} );
+
+			const fullBillingHistory = billing_history;
+			if ( billing_history.length === limit ) {
+				// If we have exactly the limit number of transactions, it means there are more transactions to fetch.
```

Comment:
Is that true? What if there happen to just be 200 receipts?

I think we probably need the endpoint to also return a total so we can tell how many there actually are.

Reply:
@sirbrillig 

> Is that true? What if there happen to just be 200 receipts?

It will make a call for nothing...

> I think we probably need the endpoint to also return a total so we can tell how many there actually are.

I know. I thought about that. It is actually already provided in the underlying `get_receipts_items_for_user` function.

However, it will require a good refactor of the back-end... I assumed it was worth the work

Reply:
Oh, I see. It will fetch more but return 0 results. Ok, that might be an acceptable compromise. We're going to need the total available to the frontend in order to perform pagination requests but I suppose this is just a first step. Ok.

Reply:
I gave it a try, but it looks like assessing the total number is actually not possible. See [this comment](https://linear.app/a8c/issue/SHILL-836/return-the-total-number-of-receipts-in-the-endpoint-result#comment-73cc0119)